File: src/main/java/smart4aviation/repository/FlightRepository.java
package smart4aviation.repository;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import smart4aviation.entity.Flight;

import java.io.IOException;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

@Service
public class FlightRepository extends JsonRepository<Flight> {

    public FlightRepository(
            @Value("${mock.flights}")
            String mockPath,
            ObjectMapper objectMapper
    ) {
        super(mockPath, new TypeReference<>() {}, objectMapper);
    }

    public List<Flight> findFlightDepartingFrom(String IATACode) throws IOException {
        return findAll().stream().filter(flight -> Objects.equals(flight.getDepartureAirportIATACode(), IATACode)).collect(Collectors.toList());
    }

    public List<Flight> findFlightArrivingTo(String IATACode) throws IOException {
        return findAll().stream().filter(flight -> Objects.equals(flight.getArrivalAirportIATACode(), IATACode)).collect(Collectors.toList());
    }

}

File: src/main/kotlin/com/example/smart4aviation/repository/FlightRepository.kt
package com.example.smart4aviation.repository

import com.example.smart4aviation.entity.Flight
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class FlightRepository(
    @Value("\${mock.flights}")
    mockPath: String,
    objectMapper: ObjectMapper
) : JsonRepository<Flight>(mockPath, object : TypeReference<List<Flight>>() {}, objectMapper) {

    fun findFlightDepartingFrom(IATACode: String): List<Flight> {
        return findAll().filter { it.departureAirportIATACode == IATACode }
    }

    fun findFlightArrivingTo(IATACode: String): List<Flight> {
        return findAll().filter { it.arrivalAirportIATACode == IATACode }
    }

}